File: semestr-5/Algorytmy i struktury danych 1/insertion_sort.py

def insertion_sort(tab):
    if len(tab) == 0:
        return tab
    for i in range(1, len(tab)):
        x = tab[i]
        j = i - 1
        while j >= 0 and tab[j] > x:
            tab[j + 1] = tab[j]
            j -= 1
        tab[j + 1] = x
        
        print(i, tab)
    return tab


tab = [11,5,7,2,15,4,6,12,87,36,22,17,91,92,96,97]

insertion_sort(tab)

File: semestr-2/Podstawy programowania/ppr/ANTy_PP/_Pomoce/insertionsort.c
/* Program sortuje tablice nierosnaco algorytmem InsertionSort.
(sortowanie przez wstawianie) */
#include <stdio.h>
#include <stdlib.h>
#define rozm 20
void zamien(int tab[], int i, int j)
{
 int temp;
temp=tab[i]; tab[i]=tab[j]; tab[j]=temp;
}
void InsertionSort(int tab[], int n)
{
 int i,j;
for(i=1;i<n;i++){
j=i;
while(j>0 && tab[j]>tab[j-1]){ //wystarczy zmienic znak '<' na '>'
zamien(tab,j,j-1); //zeby otrzymac ciag niemalejacy
j--;
}
}
}
void wyp_los(int tab[], int n, int k)
{
 int i; for(i=0;i<n;i++) tab[i]=random(k);
}
void wypisz(int tab[], int n)
{
 int i; for(i=0;i<n;i++) printf("%i, ",tab[i]);
printf("\n");
}
void main()
{
 int t[rozm];
randomize();
wyp_los(t,rozm,rozm);
wypisz(t,rozm);
InsertionSort(t,rozm);
wypisz(t,rozm);
}
